core/src/main/java/hudson/slaves/NodeProvisioner.java

```diff
@@ -172,11 +172,7 @@ public NodeProvisioner(Label label, LoadStatistics loadStatistics) {
     public void suggestReviewNow() {
         if (System.currentTimeMillis() > lastSuggestedReview + TimeUnit.SECONDS.toMillis(1)) {
             lastSuggestedReview = System.currentTimeMillis();
-            Computer.threadPoolForRemoting.submit(new Runnable() {
-                public void run() {
-                    update();
-                }
-            });
+            update();
```

Comment:
Dear God, this change took for ever to diagnose as being the root cause of the failure on the test rig... I'm reverting most of this to see if I can get back to something that tries to provision as I suspect there will be race conditions elsewhere... The move away from locks really makes problems in this code hard to debug